contracts/citycoin.clar

```diff
@@ -935,4 +935,33 @@ u113 u114 u115 u116 u117 u118 u119 u120 u121 u122 u123 u124 u125 u126 u127 u128
     (asserts! (is-eq contract-caller (var-get city-wallet)) (err ERR-UNAUTHORIZED))
     (ok (var-set city-wallet wallet-address))
   )
+)
+
+;;
+;; LIFECYCLE TRAIT FUNCTIONS
+;;
+(impl-trait .citycoin-lifecycle-trait.lifecycle-trait)
+
+(define-constant CONTRACT_VERSION "0.0.1")
```

Comment:
Do we have a proposed outline for how the versioning works? Is it just major, minor, and patch? We should probably write up something like this from stacks-blockchain:

```
X.Y.Z.A.n

X = 2 and does not change in practice unless there’s another Stacks 2.0 type event
Y increments on consensus-breaking changes
Z increments on non-consensus-breaking changes that require a fresh chainstate (akin to semantic MAJOR)
A increments on non-consensus-breaking changes that do not require a fresh chainstate, but introduce new features (akin to semantic MINOR)
n increments on patches and hot-fixes (akin to semantic PATCH)
```

Reply:
Overall it could be defined however we decide to define it 😅 
We can use commit hash, color names, movie names, or just incremental values.
I wouldn't however use a typical semantic versioning as it indicates, we have a plan for future releases. And I believe we should avoid upgrades as much and as long as it is possible, because with upgrades we are loosing main benefits of smart contracts which is code immutability.

Reply:
I vote movie names! 😆 Joking aside - was curious if you chose the 3 digits here because there was something else in mind, versus just having it be "Version 1, Version 2, and so on", e.g.

```
X.Y.Z

X = major change to contract logic, introduces new features, etc
Y = minor change to contract logic, fixes a critical bug, etc
Z = ???
```
Semantic versioning could be applied to help outline *what* type of changes were made, while we aren't planning for updates we could probably outline some scenarios where an upgrade is necessary and what that looks like, similar to SIP-011 for the stacks-blockchain.

I agree upgrades are intended to be a last resort measure and avoided when possible.

Reply:
Just muscle memory. I used `0.0.1` so many times in my career for initial version of something, that it's just a habit right now (probably a bad habit). 

You know what... Lets try with semver. People know how it works and it is well described all over the internet. 
In the end `2.1.0` is a bit more professional and welcoming than `Die Hard`

Reply:
Makes sense to me! I'll create a separate issue to define semver for our use cases but nothing that needs to be done now.

Although, `A Good Day to Die Hard` does sound like an epic bug fix!